Go client sdk built-in nats-server unable to read configuration file by passing ConfigFile Options.

### Observed behavior

```go
package main

import (
	"fmt"
	"net/url"
	"os"
	"os/exec"
	"path/filepath"
	"strings"
	"time"

	"github.com/nats-io/nats-server/v2/server"
	"github.com/nats-io/nats.go"
)

type NATSClient struct {
	url  string
	opts []nats.Option
}

func main() {
	const (
		address          = "0.0.0.0"
		mainPort         = 4222
		clusterPort      = 7422
		leafPort         = 4223
		subjectName      = "foo"
		credsFileAppUser = "~/.local/share/nats/nsc/keys/creds/local/APP/user.creds"
		credsFileSysUser = "~/.local/share/nats/nsc/keys/creds/local/SYS/sys.creds"
	)

	// Step 1: Update operator configuration
	fmt.Println("Updating operator configuration...")
	execCmd := exec.Command("nsc", "edit", "operator", "--require-signing-keys",
		"--account-jwt-server-url", fmt.Sprintf("nats://%s:%d", address, mainPort))
	output, err := execCmd.CombinedOutput()
	if err != nil {
		panic(fmt.Sprintf("Failed to update operator: %v\nOutput: %s", err, string(output)))
	}
	fmt.Printf("Operator update output:\n%s\n", string(output))

	// Step 2: Update APP account
	fmt.Println("Updating APP account configuration...")
	execCmd = exec.Command("nsc", "edit", "account", "APP", "--sk", "generate")
	output, err = execCmd.CombinedOutput()
	if err != nil {
		panic(fmt.Sprintf("Failed to update APP account: %v\nOutput: %s", err, string(output)))
	}
	fmt.Printf("APP account update output:\n%s\n", string(output))

	// Step 3: Generate resolver configuration
	fmt.Println("Generating resolver configuration...")
	execCmd = exec.Command("nsc", "generate", "config", "--nats-resolver", "--sys-account", "SYS")
	output, err = execCmd.CombinedOutput()
	if err != nil {
		panic(fmt.Sprintf("Failed to generate resolver configuration: %v\nOutput: %s", err, string(output)))
	}

	resolverFile := "resolver.conf"
	err = os.WriteFile(resolverFile, output, 0644)
	if err != nil {
		panic(fmt.Sprintf("Failed to write resolver.conf: %v", err))
	}
	fmt.Println("resolver.conf created")

	// Get NATS server configuration
	absResolverPath, _ := filepath.Abs(resolverFile)

	// Step 5: Start the main NATS server using Go library
	fmt.Println("Starting main NATS server using Go...")
	mainServerOpts := server.Options{
		ConfigFile: absResolverPath,
		Host:       address,
		Port:       mainPort,
		LeafNode: server.LeafNodeOpts{
			Host:      address,
			Port:      clusterPort,
			Advertise: fmt.Sprintf("%s:%d", address, clusterPort),
		},
		NoSigs: true, // Disable signal handling
	}

	mainServer, err := server.NewServer(&mainServerOpts)
	if err != nil {
		panic(fmt.Sprintf("Failed to create main server: %v", err))
	}

	// Start the server
	go mainServer.Start()
	defer mainServer.Shutdown()

	if !mainServer.ReadyForConnections(10 * time.Second) {
		panic("Main server not ready for connections within timeout")
	}

	fmt.Println("Main NATS server started")

	// Step 6: Test system account connection
	fmt.Println("Testing system account connection...")
	expandedSysCreds := strings.Replace(credsFileSysUser, "~", os.Getenv("HOME"), 1)
	sysClientOpt := NATSClient{
		url: fmt.Sprintf("nats://%s:%d", address, mainPort),
		opts: []nats.Option{
			nats.Name("System Client"),
			nats.ReconnectWait(time.Second),
			nats.MaxReconnects(-1),
			nats.UserCredentials(expandedSysCreds),
		},
	}

	sysConn, err := nats.Connect(sysClientOpt.url, sysClientOpt.opts...)
	if err != nil {
		panic(fmt.Sprintf("System account connection failed: %v", err))
	}
	fmt.Println("System account connected successfully")
	sysConn.Close()

	// Step 7: Push APP account JWT
	fmt.Println("Pushing APP account JWT...")
	pushCmd := exec.Command("nsc", "push", "-a", "APP", "-u",
		fmt.Sprintf("nats://%s:%d", address, mainPort))
	output, err = pushCmd.CombinedOutput()
	if err != nil {
		panic(fmt.Sprintf("Failed to push JWT: %v\nOutput: %s", err, string(output)))
	}
	fmt.Printf("JWT push output:\n%s\n", string(output))

	// Step 8: Start the leaf node NATS server using Go library
	fmt.Println("Starting leaf node NATS server using Go...")
	expandedAppCreds := strings.Replace(credsFileAppUser, "~", os.Getenv("HOME"), 1)

	leafServerOpts := server.Options{
		Host: address,
		Port: leafPort,
		LeafNode: server.LeafNodeOpts{
			ReconnectInterval: 2 * time.Second,
			Remotes: []*server.RemoteLeafOpts{
				{
					URLs:        []*url.URL{{Scheme: "nats", Host: fmt.Sprintf("%s:%d", address, clusterPort)}},
					Credentials: expandedAppCreds,
				},
			},
		},
		NoSigs: true, // Disable signal handling
	}

	leafServer, err := server.NewServer(&leafServerOpts)
	if err != nil {
		panic(fmt.Sprintf("Failed to create leaf node server: %v", err))
	}

	// Start leaf node server
	go leafServer.Start()
	defer leafServer.Shutdown()

	if !leafServer.ReadyForConnections(10 * time.Second) {
		panic("Leaf node server not ready for connections within timeout")
	}

	fmt.Println("Leaf node NATS server started")

	// Wait for leaf node connection to establish
	time.Sleep(3 * time.Second)

	// Step 9: Create main client connection and subscription
	fmt.Println("Creating main client connection...")
	mainClientOpt := NATSClient{
		url: fmt.Sprintf("nats://%s:%d", address, mainPort),
		opts: []nats.Option{
			nats.Name("Main Client"),
			nats.ReconnectWait(time.Second),
			nats.MaxReconnects(-1),
			nats.UserCredentials(expandedAppCreds),
		},
	}

	mainConn, err := nats.Connect(mainClientOpt.url, mainClientOpt.opts...)
	if err != nil {
		panic(fmt.Sprintf("Failed to connect to main server: %v", err))
	}
	defer mainConn.Close()

	mainSub, err := mainConn.Subscribe(subjectName, func(msg *nats.Msg) {
		fmt.Printf("Main client received message: %s\n", string(msg.Data))
		msg.Respond([]byte("Hello from main client!"))
	})
	if err != nil {
		panic(fmt.Sprintf("Failed to create subscription on main client: %v", err))
	}
	defer mainSub.Unsubscribe()

	// Step 10: Create leaf client connection and make request
	fmt.Println("Creating leaf client connection...")
	leafClientOpt := NATSClient{
		url: fmt.Sprintf("nats://%s:%d", address, leafPort),
		opts: []nats.Option{
			nats.Name("Leaf Client"),
			nats.ReconnectWait(time.Second),
			nats.MaxReconnects(-1),
		},
	}

	leafConn, err := nats.Connect(leafClientOpt.url, leafClientOpt.opts...)
	if err != nil {
		panic(fmt.Sprintf("Failed to connect to leaf server: %v", err))
	}
	defer leafConn.Close()

	// Wait for subscription propagation
	time.Sleep(1 * time.Second)

	// Send request and wait for response
	fmt.Println("Sending request from leaf node to main server...")
	respond, err := leafConn.Request(subjectName, []byte("Hello from leaf client!"), 5*time.Second)
	if err != nil {
		panic(fmt.Sprintf("Failed to make request from leaf client: %v", err))
	}
	fmt.Printf("Leaf client received response: %s\n", string(respond.Data))

	fmt.Println("Test completed successfully!")
	time.Sleep(1 * time.Second)
}
```
stdout :
```text
Updating operator configuration...
Operator update output:
[ OK ] set account jwt server url to "nats://0.0.0.0:4222"
[ OK ] edited operator "local"

Updating APP account configuration...
APP account update output:
[ OK ] added signing key "ADLXSSW2J44MTTJBLDDFDLHHLDCEHKILB57PHJRRWT6K2AZTMHWVZ7FH"
[ OK ] edited account "APP"

Generating resolver configuration...
resolver.conf created
Starting main NATS server using Go...
Main NATS server started
Testing system account connection...
System account connected successfully
Pushing APP account JWT...
panic: Failed to push JWT: exit status 1
	Output: [ERR ] push to nats-server "nats://0.0.0.0:4222" using system account "SYS":
	       [ERR ] push APP to nats-server with nats account resolver:
	              [ERR ] failed to get response to push account: nats: no responders available for request
	              [ OK ] pushed to a total of 0 nats-server
	Error: all jobs failed
	

goroutine 1 [running]:
main.main()
	/home/orician/workspace/learn/nats/Dchat/cmd/LeafJWT/test.go:124 +0x370b
```

### Expected behavior

Expected stdout :
```text
Updating operator configuration...
Operator update output:
[ OK ] set account jwt server url to "nats://0.0.0.0:4222"
[ OK ] edited operator "local"

Updating APP account configuration...
APP account update output:
[ OK ] added signing key "ACD4Y65UENEIO7FT3IDSNC5JKNF36CB6NR4WBTBW6AOJAKXYYUTPY3II"
[ OK ] edited account "APP"

Generating resolver configuration...
resolver.conf created
Starting main NATS server using Go...
Main NATS server started
Testing system account connection...
System account connected successfully
Pushing APP account JWT...
JWT push output:
[ OK ] push to nats-server "nats://0.0.0.0:4222" using system account "SYS":
       [ OK ] push APP to nats-server with nats account resolver:
              [ OK ] pushed "APP" to nats-server NDG3G44U5EYOG5YFU2MRKMRXU35XP72R257FFW3IAIBROJNAVAZPJ3SK: jwt updated
              [ OK ] pushed to a total of 1 nats-server

Starting leaf node NATS server using Go...
Leaf node NATS server started
Creating main client connection...
Creating leaf client connection...
Sending request from leaf node to main server...
Main client received message: Hello from leaf client!
Leaf client received response: Hello from main client!
Test completed successfully!
```

### Server and client version

go require :
github.com/nats-io/nats.go v1.43.0
github.com/nats-io/nats-server/v2 v2.11.6


### Host environment

 OS: Linux Mint 22.1 x86_64
 Kernel: Linux 6.8.0-64-generic
 CPU: AMD Ryzen 7 8745HS (16) @ 4.97 GHz

### Steps to reproduce

By execute the go program in "Observed behavior"

--- a/server/opts_test.go
+++ b/server/opts_test.go
@@ -3921,3 +3921,76 @@ func TestOptionsProxyRequired(t *testing.T) {
 	require_NoError(t, err)
 	checkUsersAndNkeys(o.LeafNode.Users, false, nil)
 }
+
+// TestNewServerFromConfigFunctionality tests the NewServerFromConfig() function
+// to ensure it properly processes config files and creates servers correctly.
+func TestNewServerFromConfigFunctionality(t *testing.T) {
+	// Test 1: Error handling - invalid configuration
+	confFileName := createConfFile(t, []byte(`
+		max_payload = 3000000000
+	`))
+
+	opts1 := &Options{
+		ConfigFile: confFileName,
+	}
+
+	// Should fail due to oversized max_payload (same as TestLargeMaxPayload)
+	if _, err := NewServerFromConfig(opts1); err == nil {
+		t.Fatalf("Expected an error from too large of a max_payload entry")
+	}
+
+	// Test 2: Config validation error - max_pending > max_payload
+	confFileName = createConfFile(t, []byte(`
+		max_payload = 100000
+		max_pending = 50000
+	`))
+
+	opts2 := &Options{
+		ConfigFile: confFileName,
+	}
+
+	// This should trigger validation error (same as TestLargeMaxPayload)
+	server, err := NewServerFromConfig(opts2)
+	if err == nil || !strings.Contains(err.Error(), "cannot be higher") {
+		if server != nil {
+			server.Shutdown()
+		}
+		t.Fatalf("Expected validation error, got: %v", err)
+	}
+}
+
+// TestNewServerFromConfigVsLoadConfig tests that NewServerFromConfig produces
+// equivalent results to the traditional LoadConfig approach.
+func TestNewServerFromConfigVsLoadConfig(t *testing.T) {
+	confFileName := createConfFile(t, []byte(`
+		port = 4224
+		max_payload = 4194304
+		max_connections = 200
+		ping_interval = "30s"
+	`))
+
+	// Method 1: Using LoadConfig (traditional approach)
+	opts1 := LoadConfig(confFileName)
+
+	// Method 2: Using NewServerFromConfig (new approach for embedded servers)
+	opts2 := &Options{ConfigFile: confFileName}
+
+	// Test 1: Both should be able to create servers successfully
+	server1, err := NewServer(opts1)
+	if err != nil {
+		t.Fatalf("Failed to create server with LoadConfig options: %v", err)
+	}
+	server1.Shutdown()
+
+	server2, err := NewServerFromConfig(opts2)
+	if err != nil {
+		t.Fatalf("Failed to create server with NewServerFromConfig: %v", err)
+	}
+	server2.Shutdown()
+
+	// Test 2: Both methods should produce equivalent results - normalize test environment fields
+	// LoadConfig sets these fields for testing, so we need to match them for fair comparison
+	opts2.NoSigs, opts2.NoLog = true, opts2.LogFile == _EMPTY_
+
+	checkOptionsEqual(t, opts1, opts2)
+}

--- a/server/server.go
+++ b/server/server.go
@@ -697,6 +697,15 @@ func New(opts *Options) *Server {
 	return s
 }
 
+func NewServerFromConfig(opts *Options) (*Server, error) {
+	if opts.ConfigFile != _EMPTY_ && opts.configDigest == "" {
+		if err := opts.ProcessConfigFile(opts.ConfigFile); err != nil {
+			return nil, err
+		}
+	}
+	return NewServer(opts)
+}
+
 // NewServer will setup a new server struct after parsing the options.
 // Could return an error if options can not be validated.
 // The provided Options type should not be re-used afterwards.

